segment.go

```diff
@@ -294,6 +312,69 @@ func (s *SegmentBase) loadFields() error {
 	return nil
 }
 
+func (s *SegmentBase) loadFieldsNew() error {
```

Comment:
Same here - name to indicate sections.

Or perhaps it'd be cleaner to declare a new struct which comes with new attributes and methods -
```
type SegmentBase struct {
    // leave as is
}

type SegmentBaseV2 struct {
    SegmentBase

    sectionFieldsIndexOffset uint64
    ...
}

func (s *SegmentBaseV2) loadSectionFields() error {

}

type Segment struct {
    SegmentBaseV2

    ...
}
```

If this can cause way too much of refactoring, you can ditch it. Thoughts?

Reply:
oh cool, thanks! I'll think more on these lines and get back to you on this.